DNN Platform/Library/Common/Utilities/FileExtensionWhitelist.cs

```diff
@@ -141,5 +141,19 @@ private IEnumerable<string> NormalizeExtensions(IEnumerable<string> additionalEx
         {
             return additionalExtensions.Select(ext => (ext.StartsWith(".") ? ext : "." + ext).ToLowerInvariant());
         }
+
+        public FileExtensionWhitelist RestrictBy(FileExtensionWhitelist parentList)
+        {
+            var filter = parentList.AllowedExtensions.ToList();
```

Comment:
You can just use `_extensions` directly, since we're inside `FileExtensionWhitelist`:
```suggestion
            var filter = parentList._extensions;
```